bin/find-lib-in-path.py

```diff
@@ -36,16 +35,17 @@ def get_ld_paths():
     # - LD_LIBRARY_PATH
     # - /etc/ld.so.cache (instead we will parse /etc/ld.so.conf)
     # - /lib, /usr/lib (or maybe /lib64, /usr/lib64)
-    paths = []
-    if "LD_LIBRARY_PATH" in os.environ:
-        paths.extend(os.environ["LD_LIBRARY_PATH"].split(":"))
+    PREFIX = os.getenv("PREFIX") # Termux & etc.
+    LDPATH = os.getenv("LD_LIBRARY_PATH") 
+    paths = ["/lib", "/usr/lib", "/lib64", "/usr/lib64"]
+    if LDPATH: 
+        paths.extend(LDPATH.split(":"))
+    if PREFIX:
+        paths.extend([PREFIX, PREFIX + "/lib", PREFIX + "/usr/lib", PREFIX + "/lib64", PREFIX + "/usr/lib64"])
```

Comment:
I just notice that you changed the order. The order is relevant though as it should reflect the actual search paths for executables.

(I also wonder now, which executables actually look at `PREFIX` for libraries?)

Reply:
> it should reflect the actual search paths for executables.
in what way it should reflect the actual search paths for executables? i don't get it :thinking:

Reply:
Executables when they load libraries (e.g. `libc.so`) will search through these paths in a well defined order.

Reply:
So should i remove `paths = ["/lib", "/usr/lib", "/lib64", "/usr/lib64"]` and re-add line `paths.extend(['/lib'...]`  ?

Reply:
You mean `LDPATH`s should be checked \extended  before `"/lib", "/usr/lib",  "/lib64", "/usr/lib64"` and so on ..?

Reply:
not sure how the order should be...

Reply:
Yes right. So imagine you have `libc.so` in multiple of those directories, and you start any executable which tries to load `libc.so`. Then the order matters. It will load the `libc.so` from the first directory. The order is well defined. And this is basically what I wrote in the comment in the beginning of the function:

> To be very correct, see man-page of ld.so.
> And here: http://unix.stackexchange.com/questions/354295/what-is-the-default-value-of-ld-library-path/354296
> Short version, not specific to an executable, in this order:
> - LD_LIBRARY_PATH
> - /etc/ld.so.cache (instead we will parse /etc/ld.so.conf)
> - /lib, /usr/lib (or maybe /lib64, /usr/lib64)

Reply:
Ok, order is fixed now :+1: